CMakeLists.txt

```diff
@@ -13,6 +13,13 @@ set(NVFUSER_THIRD_PARTY_DIR "${NVFUSER_ROOT}/third_party")
 option(NVFUSER_STANDALONE_BUILD_WITH_UCC "" OFF)
 option(NVFUSER_BUILD_WITH_ASAN "Build nvFuser with asan" OFF)
 
+include(CMakeDependentOption)
+cmake_dependent_option(NVFUSER_DISTRIBUTED "" ON
+	"$ENV{USE_DISTRIBUTED}" OFF)
```

Comment:
I'm a bit worried about dependency on `USE_DISTRIBUETD` from environment variable. Although we build pytorch from source with or without this variables, building against a pytorch wheel (with distributed package) won't have this environment variable set by default.

`USE_DISTRIBUTED` definition is inherited from linking libtorch. Is it possible to parse that from CMake side? If not, we can add something similar like this https://github.com/NVIDIA/Fuser/blob/c1ac1196ea546fdb77cf6732936aa4fd4848b209/tools/gen_nvfuser_version.py#L32-L46, which runs a `python -c 'import torch; torch._C._show_config()` subprocess to find out if libtorch has distributed package.

This can be a mid-term plan and doesn't need to be fixed now.

Reply:
strangely I don't see `USE_DISTRIBUTED` in `torch._C._show_config()` inside our container.

Reply:
I also do not see the `USE_DISTRIBUTED` inside our container.
I think we can parse it from `torch.distributed.is_available()`?

Reply:
We can use `torch._C._has_distributed()`. 

https://github.com/pytorch/pytorch/blob/200108c6e611875f9d4e3a1c7a1cec282b57f031/torch/csrc/Module.cpp#L526-L533